transformation/src/api/rest/pipelineConfigEndpoint.ts

```diff
@@ -45,102 +47,90 @@ export class PipelineConfigEndpoint {
 
     const answer = `Triggered all pipelines for datasource ${triggerRequest.datasourceId}. Executing asynchronously...`
     res.setHeader('Content-Type', 'text/plain')
-    res.writeHead(200)
-    res.write(JSON.stringify(answer))
-    res.end()
+    res.status(200).send(answer)
   }
 
   delete = async (req: express.Request, res: express.Response): Promise<void> => {
-    const configId = req.params.id
-    if (!configId) {
-      res.writeHead(400)
-      res.write('Path parameter id is missing or is incorrect')
-      res.end()
+    const configId = parseInt(req.params.id)
+    if (Number.isNaN(configId)) {
+      res.status(400).send('Path parameter id is missing or is incorrect')
       return
     }
-    await this.pipelineConfigManager.delete(+configId)
-    res.setHeader('Content-Type', 'application/json')
-    res.writeHead(204)
-    res.end()
+    await this.pipelineConfigManager.delete(configId)
+    res.setHeader('Content-Type', 'application/json') // Remove as there is no body/content
+    res.status(204).send()
   }
 
   deleteAll = async (req: express.Request, res: express.Response): Promise<void> => {
     await this.pipelineConfigManager.deleteAll()
-    res.writeHead(204)
-    res.end()
+    res.status(204).send()
   }
 
   update = async (req: express.Request, res: express.Response): Promise<void> => {
-    const configId = req.params.id
-    if (!configId) {
-      res.writeHead(400)
-      res.write('Path parameter id is missing or is incorrect')
-      res.end()
+    const configId = parseInt(req.params.id)
+    if (Number.isNaN(configId)) {
+      res.status(400).send('Path parameter id is missing or is incorrect')
       return
     }
-    const config = req.body as PipelineConfig
-    if (!config.transformation) {
-      config.transformation = { func: 'return data;' }
+    const validator = new PipelineConfigDTOValidator()
+    if (!validator.validate(req.body)) {
+      res.status(400).json({ errors: validator.getErrors() })
+      return
     }
+    const config = req.body
     try {
-      await this.pipelineConfigManager.update(+configId, config)
+      await this.pipelineConfigManager.update(configId, config)
     } catch (e) {
       res.status(404).send(`Could not find config with id ${configId}: ${e}`)
+      return
     }
-    res.setHeader('Content-Type', 'application/json')
-    res.writeHead(204)
-    res.end()
+    res.setHeader('Content-Type', 'application/json') // Remove as there is no body/content
```

Comment:
What is the purpose of this comment?
If the header can be removed -> remove it
if not (I guess because of tests) -> create an issue and remove it later